arch/arc/core/isr_wrapper.S

```diff
@@ -230,47 +230,38 @@ From RIRQ:
  */
 
 SECTION_FUNC(TEXT, _isr_wrapper)
+#if CONFIG_ARC_FIRQ
+#if CONFIG_RGF_NUM_BANKS == 1
+	st r0,[saved_r0]
+#endif
+
 #ifdef CONFIG_EXECUTION_BENCHMARKING
+#if CONFIG_RGF_NUM_BANKS !=1
+	lr r0, [_ARC_V2_STATUS32]
+	and r0, r0, ~_ARC_V2_STATUS32_RB(7)
+	kflag r0
+#endif
+/*
+ * read_timer_start_of_isr just uses several
+ * register now, no need to save all regs
```

Comment:
the read_timer_isr function is not in this module, so making assumptions on its register usage is risky. Saving the context is done later anyway, maybe move the read_timer call to that point? Or just leave in the full context save - it impacts the benchmark number, but not regular performance.